server/api/jurisdictions.py

```diff
@@ -719,3 +720,97 @@ def get_discrepancy_counts_by_jurisdiction(election: Election):
             for jurisdiction in election.jurisdictions
         }
     )
+
+
+@dataclass
+class Discrepancy:
+    audit_count: int
+    reported_count: int
+    candidate_name: str
+
+
+@dataclass
+class ContestDiscrepancies:
+    contest_id: str
+    jurisdiction_id: str
+    batch_name: str
+    discrepancies: List[Discrepancy]
+
+
+@api.route("/election/<election_id>/discrepancy", methods=["GET"])
+@restrict_access([UserType.AUDIT_ADMIN])
+def get_discrepancies_by_jurisdiction(election: Election):
+    current_round = get_current_round(election)
+    if not current_round:
+        raise Conflict("Audit not started")
+
+    discrepancies_by_jurisdiction: Dict[str, List[ContestDiscrepancies]] = defaultdict()
+
+    if election.audit_type == AuditType.BATCH_COMPARISON:
+        discrepancies_by_jurisdiction = (
+            get_batch_comparison_audit_discrepancies_by_jurisdiction(
+                election, current_round.id
+            )
+        )
+    else:
+        raise Conflict("Discrepancies are only implemented for batch comparison audits")
+
+    return jsonify(discrepancies_by_jurisdiction)
```

Comment:
A few thoughts on the data format here:
- I like that you use a dictionary with jurisdiction keys - this makes sense since they are unique
- I'd suggest using dictionaries more liberally to help indicate where there are no duplicates expected

So that might lead me to some of these ideas:

```py
# Nesting by contest
{
    jurisdiction_id: {
        contest_id: {
            batch_name: {
                choice_id: {reported_votes: int, audited_votes: int, discrepancy: int}
            }
        }
    }
}
# Nesting by batch (maybe slightly more applicable to the UI you mocked up)
{
    jurisdiction_id: {
        batch_name: {
            contest_id: {
                choice_id: {reported_votes: int, audited_votes: int, discrepancy: int}
            }
        }
    }
}
# Could also consider flipping the final level of the hierarchy, since we already have the reported_votes and audited_votes constructed
{
    jurisdiction_id: {
        batch_name: {
            contest_id: {
                reported_votes: {choice_id: int},
                audited_votes: {choice_id: int},
                discrepancies: {choice_id: int},
            }
        }
    }
}
```

Reply:
Chatted on Slack, key points: 
```
Nikhil - [This comment ](https://github.com/votingworks/arlo/pull/2026#discussion_r1823147709)in particular helped point out the question I was unsure about - I shied away from more nested dictionaries, because I felt the types might get a bit harder to follow with more nesting. But, I notice that in most places we don't define explicit return types like what I have for ContestDiscrepancies and Discrepancies, and we just notate the shape of the object that we send to the frontend via something like a comment. With the approach of a more nested dictionary, would you recommend I drop the classes I defined?

Jonah - I don’t have a strong opinion. In Arlo, since there’s a Python backend and TS frontend, there’s not as much value in defining types for everything, since we can’t share them. I usually make typed data structures on the server when I’m reusing them across functions within the backend. If it’s just to send data to the frontend, I just write a comment.
But that’s not set in stone, it’s just what I’ve been doing
```

Updating types to follow third option above, removing dataclass models